packages/create-sitecore-jss/src/templates/nextjs-xmcloud/scripts/generate-component-builder/plugins/code-extraction.ts

```diff
@@ -0,0 +1,16 @@
+import { ComponentBuilderPlugin, ComponentBuilderPluginConfig } from '..';
+import { extractComponents } from '@sitecore-jss/sitecore-jss-dev-tools';
+
+/**
+ * If consented to, sends components code for code generation learning to XMC
+ */
+class CodeExtractionPlugin implements ComponentBuilderPlugin {
+  order = 9999;
+
+  exec(config: ComponentBuilderPluginConfig) {
+    extractComponents({});
```

Comment:
Related to some other comments
```suggestion
    extractComponents();
```